### PR TITLE
[Merged by Bors] - feat: `det_vandermonde_add` and `det_vandermonde_sub`

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Vandermonde.lean [hunk 1]
```diff
@@ -154,6 +154,20 @@ theorem det_vandermonde_ne_zero_iff [IsDomain R] {n : ℕ} {v : Fin n → R} :
...
 #align matrix.det_vandermonde_ne_zero_iff Matrix.det_vandermonde_ne_zero_iff
 
+theorem det_vandermonde_eq_det_vandermonde_add {n : ℕ} (v : Fin n → R) (a : R) :
+    (Matrix.vandermonde v).det = (Matrix.vandermonde fun i ↦ v i + a).det := by
+  rw [Matrix.det_vandermonde, Matrix.det_vandermonde]
+  congr
+  ext i
+  congr
+  ext j
+  exact (add_sub_add_right_eq_sub (v j) (v i) a).symm
```
Comment (alexjbest) [inline Mathlib/LinearAlgebra/Vandermonde.lean:164]:
```suggestion
  simp [Matrix.det_vandermonde]
```

File: Mathlib/LinearAlgebra/Vandermonde.lean [hunk 2]
```diff
@@ -154,6 +154,15 @@ theorem det_vandermonde_ne_zero_iff [IsDomain R] {n : ℕ} {v : Fin n → R} :
   simp only [det_vandermonde_eq_zero_iff, Ne.def, not_exists, not_and, Classical.not_not]
 #align matrix.det_vandermonde_ne_zero_iff Matrix.det_vandermonde_ne_zero_iff
 
+theorem det_vandermonde_eq_det_vandermonde_add {n : ℕ} (v : Fin n → R) (a : R) :
+    (Matrix.vandermonde v).det = (Matrix.vandermonde fun i ↦ v i + a).det := by
+  simp [Matrix.det_vandermonde]
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Vandermonde.lean:159]: Please swap these statements around. Then they can become simp-lemmas. Also, you can then shorten the name to `det_vandermonde_add`.

File: Mathlib/LinearAlgebra/Vandermonde.lean [hunk 3]
```diff
@@ -154,6 +154,15 @@ theorem det_vandermonde_ne_zero_iff [IsDomain R] {n : ℕ} {v : Fin n → R} :
...
+
+theorem det_vandermonde_eq_det_vandermonde_sub {n : ℕ} (v : Fin n → R) (a : R) :
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Vandermonde.lean:161]: Idem dito.

File: Mathlib/LinearAlgebra/Vandermonde.lean (hunk 2 above)
Comment (mo271) [reply Mathlib/LinearAlgebra/Vandermonde.lean]: done

File: Mathlib/LinearAlgebra/Vandermonde.lean (hunk 3 above)
Comment (mo271) [reply Mathlib/LinearAlgebra/Vandermonde.lean]: done

**Comment by alexjbest** [comment]
maintainer merge

File: Mathlib/LinearAlgebra/Vandermonde.lean [hunk 4]
```diff
@@ -154,6 +154,17 @@ theorem det_vandermonde_ne_zero_iff [IsDomain R] {n : ℕ} {v : Fin n → R} :
   simp only [det_vandermonde_eq_zero_iff, Ne.def, not_exists, not_and, Classical.not_not]
 #align matrix.det_vandermonde_ne_zero_iff Matrix.det_vandermonde_ne_zero_iff
 
+@[simp]
+theorem det_vandermonde_add {n : ℕ} (v : Fin n → R) (a : R) :
+    (Matrix.vandermonde fun i ↦ v i + a).det =( Matrix.vandermonde v).det := by
```
Comment (alexjbest) [inline Mathlib/LinearAlgebra/Vandermonde.lean:159]:
```suggestion
    (Matrix.vandermonde fun i ↦ v i + a).det = (Matrix.vandermonde v).det := by
```

**Comment by alexjbest** [comment]
Actually please delegate until the last issue is fixed

**Comment by kmill** [comment]
Thanks!

bors r+

**Pull request closed without merging**